skimage/filters/_unsharp_mask.py

```diff
@@ -0,0 +1,104 @@
+    Parameters
+    ----------
+    image : (M[, N[, ..., P]][, C]) ndarray
+        Input image.
+    radius : scalar or sequence of scalars, optional
+        If a scalar is given, then its value is used for all dimensions.
+        If sequence is given, then there must be exactly one radius
+        for each dimension except the last dimension for multichannel images.
+        Note that 0 radius means no blurring, and negative values are
+        not allowed.
+    amount : scalar, optional
+        The details will be amplified with this factor. The factor could be 0
+        or negative. Typically, it is a small positive number, e.g. 1.0.
+    multichannel : bool, optional
+        If True, the last `image` dimension is considered as a color channel,
+        otherwise as spatial.
+
+    Returns
+    -------
+    output : (M[, N[, ..., P]][, C]) ndarray
+        Image with unsharp masking applied. Has the same shape and dtype
+        as `image`. If `image` dtype was integer, and it cannot represent
+        the values (e.g. overflow in uint8), then the values will be
+        clipped at the limits of the dtype.
+
+    Notes
+    -----
+    Unsharp masking is an image sharpening technique. It is a linear image
+    operation, and numerically stable, unlike deconvolution which is an
+    ill-posed problem.
+
+    The sharp details are identified as the difference between the original
+    image and its blurred version. These details are added back to the
+    original image after a scaling step:
+
+        enhanced image = original + amount * (original - blurred)
+
+    The technique is described in most of the digital image processing books.
+    This implementation is based on Digital Image Processing[1]_.
+
+    Examples
+    --------
+    >>> array = np.ones(shape=(5,5), dtype=np.uint8)*100
+    >>> array[2,2] = 120
+    >>> array
+    array([[100, 100, 100, 100, 100],
+           [100, 100, 100, 100, 100],
+           [100, 100, 120, 100, 100],
+           [100, 100, 100, 100, 100],
+           [100, 100, 100, 100, 100]], dtype=uint8)
+    >>> unsharp_mask(array, radius=0.5, amount=2)
+    array([[ 99,  99,  99,  99,  99],
+           [ 99,  99,  96,  99,  99],
+           [ 99,  96, 135,  96,  99],
+           [ 99,  99,  96,  99,  99],
+           [ 99,  99,  99,  99,  99]], dtype=uint8)
+
+    References
+    ----------
+    .. [1]  Jayaraman, S., Esakkirajan, S., Veerakima, T.
+            "Digital image processing", (2009), ed iii., page 274,
+            ISBN 10: 0070144796  ISBN 13: 9780070144798
+    .. [2]  Wikipedia. Unsharp masking
+            https://en.wikipedia.org/wiki/Unsharp_masking
+
+    """
+
+    if multichannel:
+        result = np.empty_like(image)
+        for channel in range(image.shape[-1]):
+            result[..., channel] = _unsharp_mask_single_channel(
+                image[..., channel], radius, amount)
+        return result
+    else:
+        return _unsharp_mask_single_channel(image, radius, amount)
```

Comment:
How will this handle RGB images?

Reply:
RGB images would be processed channel by channel. In most applications cases, this is quite fine. 
Some color space use linear combination of the RGB channels for lightness/brightness/value channel. If you convert into this space, process this channel, and convert it back, than you get approximately the same result.

Secondly, there is no perfect color space. I would not use automatic conversion inside the filters. Or if this is the decision, then make it universal everywhere in skimage. 
Maybe a decorator/wrapper could handle the conversions back and forth. 

But I think this is something what the user should do. (Does any of the function do such automatic conversion before/after processing?)

For instance, I have an RGB image, the filter converts it to HSV, process the V channel, and then?
Should it convert it back to RGB, leave it in HSV, or process it further to something else?

What about different multichannel images? E.g. 2 or 4 channels (RGBA)? If I have to choose, I would rather go for single channel version only than a multichannel with built-in color conversion, but this is just me. I would argue for consistency, and I don't see this conversion as a consistent solution across skimage. (And still: if this is the idea, then let's go for a wrapper, and do not implement it for every filter individually.)

Reply:
>Maybe a decorator/wrapper could handle the conversions back and forth.

@dvolgyes take a look at the discussion in https://github.com/scikit-image/scikit-image/pull/2240#issuecomment-249618562 (starting from my message).

Reply:
In cases where the correct option is not clear, I agree that taking a conservative approach is good.  But, this is also an excellent opportunity to guide the user, by having the notes of the docstring describe a suggested way of handling color images.

Reply:
So what is the conclusion here? Docstring update, or is there a general agreement about color management, like adding an option to 'convert2ycbcr', like in #2240? 

And what is the correct way of handling ndarray which have more than 3 or less than 3 color channels, but convert2ycbcr is enabled?

What is the correct way to handle color channels with 16/32bit integer types when convert2ycbcr is enabled? Without the conversion, one can process the image without issues. With conversion I have to choose a dynamic range. For 8 bit, this is 0-255, for float this is 0-1. What about 16 bit where the brightest pixel have  a pixel value of 192? 

Or should I just pass the whole image to the conversion function, and it will raise error, when it cannot do the conversion?

Finally, the code can handle arbitrary dimensions, including e.g. 4 'spatial' and a color dimension. What about this case? Should I perform an YCbCr conversion, do the unsharp masking, and convert it back? (It is a little bit of numpy indexing magic, but it can be done in a single line.) Or should the code complain about the wrong dimensions? (Or should I just pass it to the color conversion function, and it will complain anyway.)

Reply:
Let's just stick to non-color images for now, and a note in the docstring on a recommended approach (convert to lab or hsv and operate on l or h).

Reply:
Assuming that it passes the CI tests, I have no further plans in this PR. Please, let me know if you like the PR in current form, or there are parts which should be improved.

Reply:
Thanks, I've added some minor tweaks (that I'm happy to also submit in the form of a PR), and mentioned a few old comments that may or may not have been addressed.

Reply:
I'd keep the current approach. There is absolutely no reason to drop the already implemented support for multichannel images unless it is not in line with the literature.
I'd not add `convert2ycbcr` parameter for now. Just mentioning the preferred way of using the function (pre-converting to ycbcr) should do.